compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1243,6 +1252,21 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         }
     }
 
+    fn dectect_wrong_primitive(path: &[Segment]) -> Option<Symbol> {
+        let name = path[path.len() - 1].ident.name;
```

Comment:
I think it would be better to access this as a string to avoid adding extra symbols just for this diagnostic.
```suggestion
        let name = path[path.len() - 1].ident.name.as_str();
```

Reply:
I think having the strings pre-interned would be faster here. String interning is done with a `FxHashMap` in O(1) time, while calling `as_str` requires locking the interner.

Reply:
I think in this instance it doesn't really matter: we're only extracting a string once, and it's only when we've already hit an error, so it's not in a hot path.